src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
```

Comment:
it would be better to use a temporary FileInfo array and not write out to logFiles side-effect until we know that all of it succeeded in the temporary array.  That way, at the end, it can unconditionally always free the temporary array.